python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -104,10 +104,13 @@ def asset(
             but do not pass an input to the asset.
         metadata (Optional[Dict[str, Any]]): A dict of metadata entries for the asset.
         required_resource_keys (Optional[Set[str]]): Set of resource handles required by the op.
-        io_manager (Optional[Union[Mapping[str, IOManagerDefinition], str, IOManagerDefinition]]): The key, definition, or definition and key of
-            the IOManager used for storing the
-            output of the op as an asset, and for loading it in downstream ops
-            (default: "io_manager").
+        io_manager_key (Optional[str]): The resource key of the IOManager used
+            for storing the output of the op as an asset, and for loading it in downstream ops (default: "io_manager"). Only one of io_manager_key and io_manager_def can be provided
+        io_manager_def (Optional[Union[Mapping[str, IOManagerDefinition], IOManagerDefinition]]): The definition of the IOManager used for
```

Comment:
This mapping thing feels a little gnarly to me. Do we need to add this right now? Could we punt?

Reply:
Definitely do not need